Sources/SwiftParser/Statements.swift

```diff
@@ -806,9 +842,14 @@ extension Parser.Lookahead {
   /// Returns `true` if the current token represents the start of a statement
   /// item.
   ///
+  /// - Parameters:
+  ///   - allowRecovery: Whether to attempt to perform recovery.
+  ///   - preferExpr: If `true`, if either an expression or statement could be
+  ///     parsed, returns `false` such that an expression can be parsed.
+  ///
   /// - Note: This function must be kept in sync with `parseStatement()`.
   /// - Seealso: ``Parser/parseStatement()``
-  mutating func isStartOfStatement(allowRecovery: Bool = false) -> Bool {
+  mutating func isStartOfStatement(allowRecovery: Bool = false, preferExpr: Bool = false) -> Bool {
```

Comment:
I think we have had a similar discussion before: How bad is it to not default the `preferExpr` parameter? Just from reviewing your PR, I think whether to prefer an expression is more or less a 50/50 decision and it doesn’t really make sense to default to one, so I’d prefer if every call is required to make a decision on it. It also makes it a lot harder to just forget to pass down `preferExpr`.

Reply:
It's only currently needed by 2 call sites, but yeah I guess there aren't that many call sites to begin with